From e1c7d8928dcdb2a7087149620953eb05952f7143 Mon Sep 17 00:00:00 2001
From: Jialin Zhang <zhangjialin11@huawei.com>
Date: Thu, 8 Dec 2022 21:45:43 +0800
Subject: [PATCH] sched: disable sched_autogroup by default
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e1c7d8928dcdb2a7087149620953eb05952f7143
Modified-by-SEL: No


hulk inclusion
category: performance
bugzilla: 32059, https://gitee.com/openeuler/kernel/issues/I65DOZ
CVE: NA

--------------------------------

This option optimizes the scheduler for common desktop workloads by
automatically creating and populating task groups.  This separation
of workloads isolates aggressive CPU burners (like build jobs) from
desktop applications.  Task group autogeneration is currently based
upon task session.

We do not need this for mostly server workloads, so just disable by
default. If you need this feature really, just enable it by sysctl:

sysctl -w kernel.sched_autogroup_enabled=1

Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/autogroup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/autogroup.c b/kernel/sched/autogroup.c
index 2067080bb235..bcb2bb80919a 100644
--- a/kernel/sched/autogroup.c
+++ b/kernel/sched/autogroup.c
@@ -5,7 +5,7 @@
 #include <linux/nospec.h>
 #include "sched.h"
 
-unsigned int __read_mostly sysctl_sched_autogroup_enabled = 1;
+unsigned int __read_mostly sysctl_sched_autogroup_enabled;
 static struct autogroup autogroup_default;
 static atomic_t autogroup_seq_nr;
 
-- 
2.33.0

